File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        
        sort(arr.begin(), arr.end());
        
        unordered_set<int> s; // delta -> index;
        int delta = arr[1]-arr[0];
        s.insert(delta);
        
        for(int i=2; i<arr.size(); i++){
            
            delta = arr[i] - arr[i-1];
            
            if(s.find(delta) != s.end()){
                continue;
            }
            else{
                return false;
            }
        }
        
        return true;
    }
};

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.java
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        
        int n = arr.length;
        Arrays.sort(arr);
        
        int difference = arr[1] - arr[0];
        
        for(int i=2; i<n; i++){
            
            int curDifference = arr[i] - arr[i-1];
            if(curDifference != difference)
                return false;
        }
        
        return true;
    }
}